test/creds_test.go

```diff
@@ -430,19 +428,25 @@ func (s) TestCredsHandshakeAuthority(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer lis.Close()
```

Comment:
I think calling `s.Stop()` below would automatically close the listener. This may be redundant. Same comment for other places listener is being closed explicitly.

Reply:
understood reverted the change.